### DISCUSSION TITLE
Tables not dropped after mikro-orm schema:fresh

### DISCUSSION THREAD

**Discussion by simon-schuster** [mikro-orm/mikro-orm#4615] [Q&A] [answered]
**Describe the bug**
When deleting an entity and then trying to recreate the database, the tables for the deleted entities are not dropped.
Also observed this behaviour when just trying to drop the current schema.

**Stack trace**
```
Executing task: yarn run database:create 

yarn run v1.22.19
$ mikro-orm schema:fresh --run
Schema successfully dropped and recreated
Done in 2.12s.
Terminal will be reused by tasks, press any key to close it. 
```

**To Reproduce**
1. Create a entity
2. Setup schema
3. Delete entity
4. Try running mikro-orm schema:fresh
5. Check database (deleted entity table should still exist)

**Expected behavior**
After mikro-orm schema:fresh only tables for existing entities should exist.
When using mikro-orm schema:drop, no tables should exist anymore.

**Additional context**
I'm using PostgreSQL.

**Versions**

| Dependency | Version |
| - | - |
| node | 18.15.0 |
| typescript | 5.1.6 |
| mikro-orm | 5.7.14 |
| mikro-orm/postgresql | 5.7.14 |

**Comment by B4nan** [comment, marked as answer]
You can drop the whole database with `--drop-db` flag if you want to clear it completely, this flag has been also added to the `schema:fresh` recently (you'd have to use dev version to have that, it's not released as stable yet). 

The `schema:drop` removes only tables that are mapped to entities, this is on purpose, I don't think that wiping everything would be a good default, people often map only something to entities, and changing that would be breaking. With that said, the `schema:update` will probably wipe those tables, to maybe it would make sense to align the behavior in v6 (but we would have to add some more options to control it).

**Reply by simon-schuster** [reply]
Thank you for the response. I have updated to `6.0.0-dev.90` (should be newest as of now?), but `schema:fresh` with `--drop-db` is not supported there.

But `schema:update` works, thank you very much.

`schema:create` doesn't create a new database if it doesn't exist yet. Is that also intended this way?

**Reply by B4nan** [reply]
> Thank you for the response. I have updated to 6.0.0-dev.90 (should be newest as of now?), but schema:fresh with --drop-db is not supported there.

This is not yet rebased to v6, those changes are only in the latest v5 dev version.

> schema:create doesn't create a new database if it doesn't exist yet. Is that also intended this way?

Yes, that is intentional, but using the schema generator itself should ensure the db exists IIRC. Haven't checked if the `--drop-db` flag works correctly with the `fresh` command, maybe that will need to be handled explicitly.
